bind: prevent mismatch of bind-libs version

matjon opened the pull request:
When upgrading packages manually, like:

        opkg install bind-dig

the bind-libs package is not upgraded, which results in problems when running the program, for example:

        root@OpenWrt:~# dig
        Error loading shared library libisc-9.20.10.so: No such file or directory (needed by /usr/bin/dig)
        Error loading shared library libdns-9.20.10.so: No such file or directory (needed by /usr/bin/dig)
        Error loading shared library libisccfg-9.20.10.so: No such file or directory (needed by /usr/bin/dig)
        Error relocating /usr/bin/dig: cfg_map_getname: symbol not found
        Error relocating /usr/bin/dig: irs_resconf_getndots: symbol not found
        Error relocating /usr/bin/dig: isc_managers_destroy: symbol not found
        Error relocating /usr/bin/dig: dns_fixedname_init: symbol not found
        Error relocating /usr/bin/dig: isc_nm_read: symbol not found
        Error relocating /usr/bin/dig: dns_rdata_init: symbol not found
        Error relocating /usr/bin/dig: isc_random_uniform: symbol not found
        [...]

This has happened to me twice on 24.10.

To fix this, enforce that the version of bind-libs matches the version of any dependent packages. Use the same approach as in net/knot/Makefile: make the dependency be present twice, once in DEPENDS, the other one in EXTRA_DEPENDS.

---

## 📦 Package Details

**Maintainer:** @nmeyerhans 

**Description:**
bind: prevent mismatch of bind-libs version

---

## 🧪 Run Testing Details

- **OpenWrt Version:** SNAPSHOT (r30114-9b777547be) 
- **OpenWrt Target/Subtarget:** x86/64
- **OpenWrt Device:** Generic x86/64

---

## ✅ Formalities

- [x] I have reviewed the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

nmeyerhans commented on net/bind/Makefile:
```diff
@@ -52,6 +52,7 @@ define Package/bind/Default
   SECTION:=net
   CATEGORY:=Network
   DEPENDS:=+bind-libs +@OPENSSL_WITH_EC
+  EXTRA_DEPENDS:=bind-libs (=$(PKG_VERSION)-r$(PKG_RELEASE))
```
Does this need to be in `EXTRA_DEPENDS`, or can the version constraint be added in `DEPENDS`?

matjon commented:
Hello,

I have updated and rebased the MR. I have realized that other internal bind dependencies also need matching package versions. So, in the newer MR, I have added the `EXTRA_DEPENDS` variables to defines for `bind-server-filter-aaaa` and `bind-tools`.

Also, I have retested on latest snapshot (r30900-dcf6e9b0ed).

matjon commented:
Hello,

I have rebased, updated and retested this change (on snapshot r32406-00e6c18a93). I would like to ask you to look at it and see if it can now be merged.

Cc: @nmeyerhans @pprindeville 

By the way: merry Christmas and happy new year!
